src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -33,15 +59,14 @@ public object CreateController(ActionContext actionContext)
                     nameof(actionContext));
             }
 
-            var controller = _typeActivator.CreateInstance(
-                _serviceProvider,
-                actionDescriptor.ControllerTypeInfo.AsType());
-
-            _controllerActivator.Activate(controller, actionContext);
+            var controllerType = actionDescriptor.ControllerTypeInfo.AsType();
+            var controller = _controllerActivator.Create(actionContext, controllerType);
+            Activate(controller, actionContext);
```

Comment:
If we're changing the terminology for the thing that instantiates the controller, we should change this as well. Personally I think the naming of all of these things is backwards.

It's wierd to see `_activator.Create(...)` followed by `Activate(...)` maybe `ActivateProperties(...)`?